### DISCUSSION TITLE
Calculate execution time of a function/route

### DISCUSSION THREAD

**Discussion by cheskoxd** [pocketbase/pocketbase#4676] [Q&A] [answered]
So, I was trying to calculate the execution time of a function in `pb_hooks/main.pb.js` using `console.time` , but I got this error when executing:

```
ERROR POST /a-route
└─ Something went wrong while processing your request.
   └─ TypeError: Object has no member 'time' at <eval>:13:17(32)
```
Code:
``` js
console.time("fn")
const fn= () =>{ /* some function */}
console.timeEnd("fn") 
```
So I found comment https://github.com/pocketbase/pocketbase/discussions/3024#discussioncomment-6591840 that explains that `console.*` uses the default Go logger for everything, and im not an expert in Go, but I couldnt find an equivalent to `console.time` 

Then I tried with `performance.now()` and I got the same error:

```
ERROR POST /a-route
└─ Something went wrong while processing your request.
   └─ ReferenceError: performance is not defined at <eval>:14:17(29)
```

So at the end I found that I can use the [DateTime ](https://pocketbase.io/jsvm/classes/DateTime.html) object like so 

```js 
const dt0 = new DateTime().time().unixMilli()
const fn= () =>{ /* some function */}
const dt1 = new DateTime().time().unixMilli()
console.log("Time: " + dt1 - dt0)
```

but I still dont feel that this is the correct way of doing it, is there a way of doing something like `console.time` ?

**Comment by ganigeorgiev** [comment, marked as answer]
The [goja `console` module](https://github.com/dop251/goja_nodejs/tree/master/console) doesn't implement `console.time`.

What is wrong with the dates substractions? If you want a slightly simpler alternative you could try with `Date.now()`:

```js
const start = Date.now();
yourfunc()
console.log("took:", Date.now() - start)
```

Another alternative could be to enable `$apis.activityLogger($app)` middleware for your route and you can find an `execTime` field in the _Admin UI > Logs_ section.

**Reply by cheskoxd** [reply]
Oh, I didn't try with `Date.now()` since I found [this one](https://pocketbase.io/jsvm/classes/DateTime.html) on jsvm I though that `Date.now()` woudln't work 😅 

Thanks a lot for the fast reply!
And sorry, next time Ill create a discussion instead 😅
